paramiko/agent.py

```diff
@@ -407,12 +410,22 @@ def __str__(self):
     def get_name(self):
         return self.name
 
-    def sign_ssh_data(self, data):
+    def sign_ssh_data(self, data, flags=0):
+        """
+        Sign a blob of data with this private key, and return a `.Message`
+        representing an SSH signature message.
+
+        Use flags to sign using sha2-256 or sha2-512, default is sha1
```

Comment:
default is sha1 only for rsa keys, not for ecdsa or ed25519, right?

Reply:
Yes. flags are not defined for anything but ssh-rsa.